hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -161,7 +160,8 @@ protected AbstractHoodieLogRecordReader(HoodieStorage storage, String basePath,
     this.latestInstantTime = latestInstantTime;
     this.hoodieTableMetaClient = hoodieTableMetaClientOption.orElseGet(
         () -> HoodieTableMetaClient.builder()
-            .setConf(storage.getConf().newInstance()).setBasePath(basePath).build());
+            .setStorage(storage.newInstance(new StoragePath(basePath), storage.getConf().newInstance()))
```

Comment:
may be a novice question. 
FileSystem or HoodieStorage is generally a immutable object right. So, why do we need to create a new instance everytime?

Reply:
I understand hadoop conf or StorageConf, we need to create new instance everytime. but trying to gauge for FileSystem

Reply:
Hadoop-based `FileSystem` and `HoodieStorage` instances are tied to the file system scheme, so to be cautious this one tries to get the correct one.

Reply:
we did have some mis-steps in creating new Path() in lot of places unnecessarily which we later figured out while root causing perf issue. So, just trying to be cautious here if we really wanted a new instance everytime.

Reply:
Makes sense

Reply:
Addressed in #11309